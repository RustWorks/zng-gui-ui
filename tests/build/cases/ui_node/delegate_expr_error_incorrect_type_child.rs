use zero_ui::wgt_prelude::{ui_node, UiNode};

struct NotANode;

struct MyNode {
    child: NotANode,
}

#[ui_node(child)]
impl UiNode for MyNode {}

fn main() {}
