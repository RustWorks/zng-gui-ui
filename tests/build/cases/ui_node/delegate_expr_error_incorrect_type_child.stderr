error[E0277]: the trait bound `NotANode: zero_ui::prelude::UiNode` is not satisfied
 --> cases/ui_node/delegate_expr_error_incorrect_type_child.rs:9:11
  |
9 | #[ui_node(child)]
  | ----------^^^^^--
  | |         |
  | |         the trait `zero_ui::prelude::UiNode` is not implemented for `NotANode`
  | required by a bound introduced by this call
  |
  = help: the following other types implement trait `zero_ui::prelude::UiNode`:
